File: app/src/main/java/cat/xojan/random1/domain/entities/CrashReporter.java
package cat.xojan.random1.domain.entities;

import com.crashlytics.android.Crashlytics;

public class CrashReporter {

    private Crashlytics mCrashlytics;

    public CrashReporter() {
    }

    public CrashReporter(Crashlytics crashlytics) {
        mCrashlytics = crashlytics;
    }

    public void logException(Throwable e) {
        if (mCrashlytics != null) mCrashlytics.logException(e);
    }

    public void logException(String message) {
        if (mCrashlytics != null) mCrashlytics.log(message);
    }
}

File: app/src/main/java/cat/xojan/random1/domain/entities/CrashReporter.kt
package cat.xojan.random1.domain.entities

import com.crashlytics.android.Crashlytics


class CrashReporter(val crashlytics: Crashlytics?) {

    fun logException(e: Throwable) {
        crashlytics?.let {
            Crashlytics.logException(e)
        }
    }

    fun logException(message: String) {
        crashlytics?.let {
            Crashlytics.log(message)
        }
    }
}